text.py

```diff
@@ -8,7 +8,7 @@
 from learning import CountingProbDist
 import search
 
-from math import log, exp
+import math
```

Comment:
I think it would be best if you left it as `from math import log, exp`, unless it interferes with code.